src/Git/CreateTagViaConsole.php

```diff
@@ -6,25 +6,39 @@
 
 use Laminas\AutomaticReleases\Git\Value\BranchName;
 use Laminas\AutomaticReleases\Gpg\SecretKeyId;
-use Psl\Env;
-use Psl\File;
-use Psl\Filesystem;
-use Psl\Shell;
+
+use function Psl\Env\temp_dir;
+use function Psl\File\write;
+use function Psl\Filesystem\create_temporary_file;
+use function Psl\Shell\execute;
 
 final class CreateTagViaConsole implements CreateTag
 {
+    public function __construct(private HasTag $hasTag)
+    {
+    }
+
     public function __invoke(
         string $repositoryDirectory,
         BranchName $sourceBranch,
         string $tagName,
         string $changelog,
         SecretKeyId $keyId,
     ): void {
-        $tagFileName = Filesystem\create_temporary_file(Env\temp_dir(), 'created_tag');
+        if (($this->hasTag)($repositoryDirectory, $tagName) === true) {
```

Comment:
`=== true` to be omitted: the interface states it's a boolean return value anyway